capabilities/controller/pkg/capabilities/core/capability_controller.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package core
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/core/v1alpha1"
+	"github.com/vmware-tanzu/tanzu-framework/capabilities/client/pkg/discovery"
+	"github.com/vmware-tanzu/tanzu-framework/capabilities/controller/pkg/config"
+	"github.com/vmware-tanzu/tanzu-framework/capabilities/controller/pkg/constants"
+)
+
+// CapabilityReconciler reconciles a Capability object.
+type CapabilityReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+	Host   string
+}
+
+//+kubebuilder:rbac:groups=run.tanzu.vmware.com,resources=capabilities,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=run.tanzu.vmware.com,resources=capabilities/status,verbs=get;update;patch
+
+// Reconcile reconciles a Capability spec by executing specified queries.
+func (r *CapabilityReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	ctxCancel, cancel := context.WithTimeout(ctx, constants.ContextTimeout)
+	defer cancel()
+
+	log := r.Log.WithValues("capability", req.NamespacedName)
+	log.Info("Starting reconcile")
+
+	capability := &corev1alpha1.Capability{}
+	if err := r.Get(ctxCancel, req.NamespacedName, capability); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	var serviceAccountName, namespaceName string
+	// use the default service account when the serviceAccountName is not provided as part of the spec
```

Comment:
```suggestion
	// Use the default service account when the service account name is not provided as part of the spec.
```

Reply:
`serviceAccountName` is the field name in the spec